jekyll/_posts/guides/2016-10-18-e2e_implementation.rst

```diff
@@ -125,43 +125,27 @@ libolm. The (base64-encoded) keys are retrieved by calling
 ``olm_account_identity_keys``. The account should be stored for future
 use.
 
-It should then publish these keys to the homeserver. To do this, it
-should construct a JSON object as follows:
+It should then publish these keys to the homeserver, which is done by using
+the ``device_keys`` property of the `/keys/upload`_ endpoint.
 
-.. code:: json
-
-  {
-    "algorithms": ["m.olm.v1.curve25519-aes-sha2", "m.megolm.v1.aes-sha2"],
-    "device_id": "<device_id>",
-    "keys": {
-      "curve25519:<device_id>": "<curve25519_key>",
-      "ed25519:<device_id>": "<ed25519_key>"
-    },
-    "user_id: <user_id>"
-  }
-
-The object should be formatted as `Canonical
-JSON <http://matrix.org/docs/spec/server_server/unstable.html#canonical-json>`__,
-then signed with ``olm_account_sign``; the signature should be added to
-the JSON as ``signatures.<user_id>.ed25519:<device_id>``.
-
-The signed JSON is then uploaded via
-``POST /_matrix/client/unstable/keys/upload``.
+In order to sign the ``device_keys`` payload as described in `Signing JSON`_,
+clients should call ``olm_account_sign``.
 
 Creating and registering one-time keys
 --------------------------------------
 
-At first start, and at regular intervals
-thereafter\ [#]_, the client should check how
-many one-time keys the homeserver has stored for it, and, if necessary,
-generate and upload some more.
-
-.. [#] Every 10 minutes is suggested.
+At first start, the client should check how many one-time keys the homeserver
+has stored for it, and, if necessary, generate and upload some more.
 
 The number of one-time keys currently stored is returned by
-``POST /_matrix/client/unstable/keys/upload``. (Post an empty JSON object
+``POST /_matrix/client/r0/keys/upload``. (Post an empty JSON object
 ``{}`` if you don’t want to upload the device keys.)
 
+However, a client should not rely on this in order to find out how many
```

Comment:
It meant for performance reasons. I rephrased it.

Reply:
OK, I see.  "should not rely on" usually means that the value will be inaccurate, so I'd use different wording.

Looking at the whole section, though, it kind of seems weird to say "you can find the value by doing this, but don't do that, do this other thing instead".  So I think it would be better to just drop the mention of checking `keys/upload` for checking the number of keys.

Reply:
Agreed and made the change.